pkg/internal/types/types.go

```diff
@@ -33,7 +33,9 @@ type EnvFunc func(context.Context, *envconf.Config) (context.Context, error)
 // can be used to customized the behavior of the
 // environment. Changes to context are expected to surface
 // to caller. Meant for use with before/after feature hooks.
-type FeatureEnvFunc func(context.Context, *envconf.Config, Feature) (context.Context, error)
+// *testing.T is provided in order to provide pass/fail context to
+// features.
+type FeatureEnvFunc func(context.Context, *envconf.Config, *testing.T, Feature) (context.Context, error)
```

Comment:
I am wondering if something similar is needed for `StepFunc` where Assessments receive a copy of the currently executing `testing.T` context.

Reply:
I don't mind taking those up if needed but I will say that is just _slightly_ tangential since this was necessary as part of the before/after hooks. The step funcs dont have hooks like those but it is possible a step would want to just abort if the test is already failed.

I'll file an issue for that and address it separately if thats OK?

Reply:
@johnSchnake ok cool. No worries, that can come at a later time.